From 5ad3849bb635b6f339c4d1d5f648882e840c9189 Mon Sep 17 00:00:00 2001
From: Noah Hsu <i@nn.ci>
Date: Fri, 9 Sep 2022 20:54:11 +0800
Subject: [PATCH] fix: if use down proxy url

---
 server/handles/down.go | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/server/handles/down.go b/server/handles/down.go
index eb9d04a47..bbe0a30b3 100644
--- a/server/handles/down.go
+++ b/server/handles/down.go
@@ -52,7 +52,7 @@ func Proxy(c *gin.Context) {
 		downProxyUrl := storage.GetStorage().DownProxyUrl
 		if downProxyUrl != "" {
 			_, ok := c.GetQuery("d")
-			if ok {
+			if !ok {
 				URL := fmt.Sprintf("%s%s?sign=%s",
 					strings.Split(downProxyUrl, "\n")[0],
 					utils.EncodePath(rawPath, true),
@@ -103,7 +103,7 @@ func shouldProxy(storage driver.Driver, filename string) bool {
 // 4. proxy_types
 // solution: text_file + shouldProxy()
 func canProxy(storage driver.Driver, filename string) bool {
-	if storage.Config().MustProxy() || storage.GetStorage().WebProxy {
+	if storage.Config().MustProxy() || storage.GetStorage().WebProxy || storage.GetStorage().WebdavProxy() {
 		return true
 	}
 	if utils.SliceContains(conf.TypesMap[conf.ProxyTypes], utils.Ext(filename)) {
